BusinessLayer/JsonForwarder/JsonForwarder.h

```diff
@@ -0,0 +1,73 @@
+/**
+ *  Schulich Delta Hermes
+ *  Copyright (C) 2015 University of Calgary Solar Car Team
+ *
+ *  This file is part of Schulich Delta Hermes
+ *
+ *  Schulich Delta Hermes is free software:
+ *  you can redistribute it and/or modify it under the terms
+ *  of the GNU Affero General Public License as published by
+ *  the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  Schulich Delta Hermes is distributed
+ *  in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or
+ *  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+ *  General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General
+ *  Public License along with Schulich Delta Hermes.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  For further contact, email <software@calgarysolarcar.ca>
+ */
+
+#pragma once
+
+#include <QObject>
+#include <QScopedPointer>
+#include <QTimer>
+
+#include "I_JsonForwarder.h"
+
+#include "BusinessLayer/JsonForwarder/BatteryJsonForwarder/I_BatteryJsonForwarder.h"
+#include "BusinessLayer/JsonForwarder/FaultsJsonForwarder/I_FaultsJsonForwarder.h"
+#include "BusinessLayer/JsonForwarder/PowerJsonForwarder/I_PowerJsonForwarder.h"
+#include "BusinessLayer/JsonForwarder/VehicleJsonForwarder/I_VehicleJsonForwarder.h"
+
+class I_BatteryData;
```

Comment:
You can just forward declare the concrete and use that in the QScopedPointer. It won't change the testibility within the current setup.